### PR TITLE
GH-236 Add BigInteger and BigDecimal argument support

### LINKED ISSUE #236: Add BigInteger and BigDecimal argument support
### Helpful information:
Package with basic arguments.
image
Register argument:
image

### PR DESCRIPTION
**Pull request by BlackBaroness**
I also extended `AssertSuggest`'s contract by adding `assertNotEmpty` and `assertCorrect` functions. `assertCorrect` allows us to test every suggestion and print exact failing suggestion in case of assertion fails. Now, because of lack of such function, we can't understand which suggestion is broken.

Closes #236

### CODE REVIEW

File: litecommands-annotations/test/dev/rollczi/litecommands/annotations/argument/resolver/standard/BigDecimalArgumentTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        platform.execute("test -88.3873729363")
+            .assertSuccess(BigDecimal.valueOf(-88.3873729363));
+
+        platform.execute("test 14")
+            .assertSuccess(BigDecimal.valueOf(14));
+
+        platform.execute("test 0")
+            .assertSuccess(BigDecimal.valueOf(0));
+
+        final BigDecimal testBigInteger = BigDecimal.ZERO
```
Comment (Rollczi) [inline litecommands-annotations/test/dev/rollczi/litecommands/annotations/argument/resolver/standard/BigDecimalArgumentTest.java:37]:
```suggestion
        BigDecimal testBigInteger = BigDecimal.ZERO
```

**Review by Rollczi — CHANGES_REQUESTED** [review]
Resolve merge conflicts

**Comment by BlackBaroness** [comment]
Done

**Comment by Rollczi** [comment]
Thanks

**Pull request merged**